App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/Profile/ContactDetail.jsx

```diff
@@ -2,269 +2,296 @@
-                    value={this.state.newContact.lastName}
-                    controlFunc={this.handleChange}
-                    maxLength={80}
-                    placeholder="Enter your last name"
-                    errorMessage="Please enter a valid last name"
-                />
-                <ChildSingleInput
-                    inputType="text"
-                    label="Email address"
-                    name="email"
-                    value={this.state.newContact.email}
-                    controlFunc={this.handleChange}
-                    maxLength={80}
-                    placeholder="Enter an email"
-                    errorMessage="Please enter a valid email"
-                />
-
-                <ChildSingleInput
-                    inputType="text"
-                    label="Phone number"
-                    name="phone"
-                    value={this.state.newContact.phone}
-                    controlFunc={this.handleChange}
-                    maxLength={12}
-                    placeholder="Enter a phone number"
-                    errorMessage="Please enter a valid phone number"
-                />
-
-                <button type="button" className="ui teal button" onClick={this.saveContact}>Save</button>
-                <button type="button" className="ui button" onClick={this.closeEdit}>Cancel</button>
-            </div>
-        )
-    }
+  handleChange(event) {
+    const data = Object.assign({}, this.state.newContact)
+    data[event.target.name] = event.target.value
+    this.setState({
+      newContact: data
+    })
+  }
+
+  saveContact() {
+    try {
+      const data = Object.assign({}, this.state.newContact);
+      const { firstName, lastName, email, phone } = data;
+      const formData = { firstName, lastName, email, phone };
 
-    renderDisplay() {
-
-        let fullName = this.props.details ? `${this.props.details.firstName} ${this.props.details.lastName}` : ""
-        let email = this.props.details ? this.props.details.email : ""
-        let phone = this.props.details ? this.props.details.phone : ""
-
-        return (
-            <div className='row'>
-                <div className="ui sixteen wide column">
-                    <React.Fragment>
-                        <p>Name: {fullName}</p>
-                        <p>Email: {email}</p>
-                        <p>Phone: {phone}</p>
-                    </React.Fragment>
-                    <button type="button" className="ui right floated teal button" onClick={this.openEdit}>Edit</button>
-                </div>
-            </div>
-        )
+      const valid = this.schema.validateSync(formData);
+      this.props.controlFunc(this.props.componentId, data);
+      this.closeEdit();
+    } catch (error) {
+      TalentUtil.notification.show(error, "error", null, null);
     }
+  }
+
+  render() {
+    return (
+      this.state.showEditSection ? this.renderEdit() : this.renderDisplay()
+    )
+  }
```

Comment:
Though this one is working, you might encounter some errors/issues with this kind of structure. Instead of using **render** first and functions below. Try to use the **render** below and the functions above like the **renderEdit** and **renderDisplay**